packages/next/src/server/response-cache/index.ts

```diff
@@ -20,7 +20,10 @@ export default class ResponseCache {
 
   constructor(minimalMode: boolean) {
     this.pendingResponses = new Map()
-    this.minimalMode = minimalMode
+    // this is a hack to avoid Webpack knowing this is equal to this.minimalMode
+    // because we replace this.minimalMode to true in production bundles.
+    const minimalModeKey = 'minimalMode'
+    this[minimalModeKey] = minimalMode
```

Comment:
I'm not sure I follow why this is required, if the variable is replaced with `true` then doesn't it still have the `minimalMode` key?

Reply:
Because this becomes "true = something" which isn't valid

Reply:
I think it's more clear to use `Object.assign(this, { minimalMode })`.